add translations for macOS UI permissions (#28)
rage/programming-20: data/macos-ohjeet.md

information_page: true
---

<!-- ## Käyttöliittymätestit -->
## User interface tests

<!-- Osa käyttöliittymätesteistä odottaa, että NetBeans/tmcbeans saa siirtää hiiren kursoria näytöllä, esimerkiksi testatakseen tehtävässä tehdyn käyttöliittymän nappuloita. macOS ei oletuksena salli turvasyistä ohjelmia siirtämään hiirtä, joten tmcbeans:lle pitää käydä lisäämässä oikeus tätä varten. -->

> Some of the user interface testing expects that Netebans/tmcbeans is allowed to move the mouse cursor on the screen. For example to test the buttons in the UI that is created in the exercise. MacOS does not allow this because of the security issues. This is why tmcbeans has to be given the permission manually.

Reviewer: ```suggestion
Some of the user interface tests expect that Netbeans/tmcbeans is allowed to move the mouse cursor on the screen. For example to test the buttons in the UI that are created in the exercise. MacOS does not allow this because of the security concerns. This is why tmcbeans has to be given the permission manually.
```

1. Open the `System Preferences`. You can open it from example by searching from Spotlight (magnifying glass on the upper right corner) by keyword _System Preferences_ and by opening the `System Preferences.app`.

![spotlight haku](./img/macos-guide/search_system_preferences.png)

<!-- 2. Paina avautuvassa näkymässä yläriviltä löytyvää Security & Privacy.
![system preferences näkymä](./img/macos-guide/system_preferences.png) -->
2. Press the `Security & Privacy` from the opened menu.

<!-- 3. Navigoi näkymässä Privacy-välilehdelle ja siellä Accessibility -osioon.
![privacy2accessibility näkymä](./img/macos-guide/navigation.png) -->
3. From there navigate to `Privacy` tab and from there to `Accessability section`.

<!-- 4. Paina vasemmasta alakulmasta "Click the lock to make changes." Username-kentässä pitäisi lukea automaattisesti tietokoneesi käyttäjätunnus. Syötä se, jos se puuttuu. Syötä salasanasi Password-kenttään ja paina Unlock.
![salasanansyöttämisnäkymä](./img/macos-guide/open-lock.png) -->

> 4. Press from the lower left corner "Click the lock to make changes.". In the username field there should be the name of your computer. Input that if it's not there. Input your password to password field and press Unlock.

Reviewer: ```suggestion
4. Press from the lower left corner "Click the lock to make changes.". In the username field there should be the name of your computer. Write if it's not there. Give your password in the password field and click Unlock.
```

<!-- 7. Paina nyt plus-kuvaketta. Navigoi avautuvassa näkymässä siihen kansioon, mihin tmcbeans on asennettuna koneellasi. Valitse tmcbeans.app ja paina Open. -->

> 7. Press now the plus-icon. Navigate to the folder where your tmcbeans is installed. Select tmcbeans.app and press Open.

Reviewer: ```suggestion
7. Click the plus-icon. Navigate to the folder where your tmcbeans is installed. Select tmcbeans.app and click Open.
```

> ![Finder-hakemisto, jossa navigoitu tmcbeansiin](./img/macos-guide/add_tmcbeans.png)

Reviewer: ```suggestion
![Finder navigated to the tmcbeans](./img/macos-guide/add_tmcbeans.png)
```

> ![accessibility-näkymä, johon on lisätty tmcbeans](./img/macos-guide/tmcbeans_added.png)

Reviewer: ```suggestion
![accessibility-view with tmcbeans added](./img/macos-guide/tmcbeans_added.png)
```

<!-- 9. Paina vasemmasta alakulmasta "Click the lock to prevent further changes." -->

> 9. Press from the lower left corner "Click the lock to prevent further changes."

Reviewer: ```suggestion
9. In the lower left corner, click "Click the lock to prevent further changes."
```

> ![accessibility-näkymä, johon ei voi tehdä enää muutoksia](./img/macos-guide/done.png)

Reviewer: ```suggestion
![accessibility-view, that doesn't allow more modifications](./img/macos-guide/done.png)
```
